### ISSUE TITLE
[BUG] evaluate does not return result with numeric error_score when estimator fitting fails for multiple metrics

### ISSUE THREAD

**Issue by hazrulakmal** [sktime/sktime#5194]
**Describe the bug**

Relates to #5167 

**To Reproduce**

```python
from sktime.datasets import load_airline
from sktime.forecasting.exp_smoothing import ExponentialSmoothing
from sktime.forecasting.model_selection import SlidingWindowSplitter
from sktime.forecasting.model_evaluation import evaluate
from sktime.performance_metrics.forecasting import (
    MeanAbsoluteError,
    MeanAbsolutePercentageError
)

scores = [MeanAbsoluteError(), MeanAbsolutePercentageError()]
forecaster = ExponentialSmoothing(sp=12)
y = load_airline()
# add NaN to make ExponentialSmoothing fail
y.iloc[1] = np.nan
fh = [1, 2, 3]
cv = SlidingWindowSplitter(step_length=33, window_length=36, fh=fh)

results = evaluate(
    forecaster=forecaster,
    y=y,
    cv=cv,
    # expected behaviour should be the same as replacing scoring with None for single metric, 
    scoring=scores 
)
```

**Output**
`FitFailedWarning` and `TypeError` is raised as below
<details>
TypeError: {'pd.DataFrame': "y_pred must be a pandas.DataFrame, found ", 'pd.Series': "y_pred must be a pandas.Series, found ", 'np.ndarray': "y_pred must be a numpy.ndarray, found ", 'xr.DataArray': "y_pred must be a xarray.DataArray, found ", 'dask_series': "y_pred must be a dask DataFrame, found ", 'df-list': "y_pred must be list of pd.DataFrame, found ", 'numpy3D': "y_pred must be a numpy.ndarray, found ", 'pd-multiindex': "y_pred must be a pd.DataFrame, found ", 'nested_univ': "y_pred must be a pd.DataFrame, found ", 'dask_panel': "y_pred must be a dask DataFrame, found ", 'pd_multiindex_hier': "y_pred must be a pd.DataFrame, found ", 'dask_hierarchical': "y_pred must be a dask DataFrame, found "}
</details>

**Expected behavior**

The `TypeError` should not be raised and the function should still return dataframe results with four rows, where the first row is filled with np.nan while the other rows are filled with corresponding values, as long as the fit does not fail. The behavior should be the same as when a single metric is supplied.

**Versions**
bug is reproducible in the main branch

**Comment by fkiraly** [comment]
I think this is simply a "sub-bug" of #5167 that should be fixed with it...

**Comment by hazrulakmal** [comment]
Yes, the reason for the failure is the same. Since the failed fit estimator cannot produce a `y_pred`, it falls back to `np.nan`. then when we compute other metrics at the end in the reformatting section, it ends up computing `np.nan` with `y_test`, which causes the break.

**Comment by hazrulakmal** [comment]
fixed at #5192
